File: src/programmers/Programmers_181928_AppendingNum.py
def solution(num_list):
    even = []
    odd = []
    for i in num_list:
        if i % 2 == 0:
            even.append(str(i))
        else:
            odd.append(str(i))

    even_str = ''.join(even)
    odd_str = ''.join(odd)

    answer = int(even_str) + int(odd_str)
    return answer


print(solution([3, 4, 5, 2, 1]))

File: src/programmers/Programmers_181928_AppendingNum.java
package programmers;

public class Programmers_181928_AppendingNum {
    public static int solution(int[] num_list) {
        StringBuilder even = new StringBuilder();
        StringBuilder odd = new StringBuilder();

        for (int i : num_list) {
            if (i % 2 == 0) {
                even.append(i);
            } else {
                odd.append(i);
            }
        }

        int evenSum = Integer.parseInt(even.toString());
        int oddSum = Integer.parseInt(odd.toString());

        return evenSum + oddSum;
    }

    public static void main(String[] args) {
        int[] numList = {3, 4, 5, 2, 1};
        int result = solution(numList);
        System.out.println(result);
    }
}
